src/pbkdf2.rs

```diff
@@ -190,17 +203,34 @@ pub fn derive(prf: &'static PRF, iterations: usize, salt: &[u8], secret: &[u8],
 ///
 /// `verify` panics if `iterations < 1`.
 ///
-/// `verify` panics if `out.len()` is larger than the output length of the
-/// digest function used by the PRF algorithm.
+/// `derive` panics if `out.len()` is larger than (2**32 - 1) * the PRF digest
+/// length, per the PBKDF2 specification
 pub fn verify(prf: &'static PRF, iterations: usize, salt: &[u8], secret: &[u8],
               previously_derived: &[u8]) -> Result<(), ()> {
     let mut derived_buf = [0u8; digest::MAX_OUTPUT_LEN];
-    if previously_derived.len() > derived_buf.len() {
-        return Err(());
+
+    let output_len = prf.digest_alg.output_len;
+    let secret = hmac::SigningKey::new(prf.digest_alg, secret);
+    let mut idx: u32 = 0;
+
+    let mut result = Ok(());
+
+    for previously_derived_chunk in previously_derived.chunks(output_len) {
+        idx = idx.checked_add(1).expect("derived key too long");
+
+        let derived_chunk = &mut derived_buf[..previously_derived_chunk.len()];
+        polyfill::slice::fill(derived_chunk, 0);
+
+        derive_block(&secret, iterations, salt, idx, derived_chunk);
+        let compares_equal = constant_time::verify_slices_are_equal(derived_chunk,
+                                                                    previously_derived_chunk);
+
+        if result.is_ok() {
```

Comment:
To avoid this branch, we need to have an internal version of `verify_slices_are_equal` that returns a `c::size_t` and then use `constant_time_select_size_t(old, new, old)` to combine all the results together. That means we have to expose `constant_time_select_size_t` to Rust, which we don't do yet.

Then, we need to map the `c::size_t` to a `Result`. But, this doesn't need to be constant time.

Let me know if you feel comfortable making these changes. If not, I'll find some time to do it soon.